primitives/message-lane/src/source_chain.rs

```diff
@@ -16,14 +16,26 @@
 
 //! Primitives of message lane module, that are used on the source chain.
 
-use crate::{InboundLaneData, LaneId};
+use crate::{InboundLaneData, LaneId, MessageNonce};
 
-use frame_support::Parameter;
-use sp_std::fmt::Debug;
+use frame_support::{Parameter, RuntimeDebug};
+use sp_std::{collections::btree_map::BTreeMap, fmt::Debug};
 
 /// The sender of the message on the source chain.
 pub type Sender<AccountId> = frame_system::RawOrigin<AccountId>;
 
+/// Relayers rewards, grouped by relayer account id.
+pub type RelayersRewards<AccountId, Balance> = BTreeMap<AccountId, RelayerRewards<Balance>>;
+
+/// Single relayer rewards.
+#[derive(RuntimeDebug, Default)]
+pub struct RelayerRewards<Balance> {
+	/// Total rewards that are needs to be paid to the relayer.
```

Comment:
```suggestion
	/// Total reward that needs to be paid to the relayer.
```

Reply:
@svyatonik please address at least this one, "that are needs" is plain incorrect.